runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -231,22 +232,37 @@ protected void dismissDialog() {
     }
 
     private void getXWalkLibrary() {
-        try {
-            Intent intent = new Intent(Intent.ACTION_VIEW);
-            startActivity(intent.setData(Uri.parse(XWALK_CORE_MARKET_URL)));
-            showDialog(getMarketProgressDialog());
-        } catch (ActivityNotFoundException e) {
-            Log.d(TAG, "Market open failed");
-            showDialog(getMarketOpenFailedDialog());
-            // TODO(sunlin): enable direct download in the future
-            // downloadXWalkLibrary();
```

Comment:
I'd suggest add comment here to explain how mXWalkApkDownloadUrl get, and when to use the defined url, when to go to Google Play Store.